db/migrate/20170829193038_create_user_preferences.rb

```diff
@@ -0,0 +1,11 @@
+class CreateUserPreferences < ActiveRecord::Migration
+  def change
+    create_table :user_preferences do |t|
+      t.references :user, index: true, foreign_key: true
```

Comment:
Would a uniqueness constraint/validation on user_id/name be a good idea?

Reply:
Added.